sockeye/data_io.py

```diff
@@ -463,9 +466,13 @@ def load(self,
             # Once MXNet allows item assignments given a list of indices (probably MXNet 1.0): e.g a[[0,1,5,2]] = x,
             # we can try again to compute the label sequence on the fly in next().
             data_label[buck_index][sample_index, :target_len] = target[1:] + [self.eos_id]
-
+            
+            map_buckets2sentence_ids[buck_index][sample_index] = sentence_id
+            
             bucket_sample_index[buck_index] += 1
 
+        self.map_buckets2sentence_ids = map_buckets2sentence_ids    
```

Comment:
thanks for the change!
You could directly operate on `self.map_buckets2sentence_ids`. One thing I would do is to add it in the constructor as optional (so that all class members are defined there) as such:
```
def __init__():
...
self.map_buckets2sentence_ids = None  # type: Optional[List[np.ndarray]]
```

Reply:
Sure, let me know if the newest commit is what you had in mind.

Reply:
looks good, thanks!